lnrpc/signrpc/signer_server.go

```diff
@@ -657,6 +682,31 @@ func (s *Server) VerifyMessage(ctx context.Context,
 	if in.Pubkey == nil {
 		return nil, fmt.Errorf("a pubkey to verify MUST be passed in")
 	}
+
+	// We allow for Schnorr signatures to be verified.
+	if in.IsSchnorrSig {
+		// We expect the public key to be in the BIP-340 32-byte format
+		// for Schnorr signatures.
+		pubkey, err := schnorr.ParsePubKey(in.Pubkey)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse pubkey: %v",
+				err)
+		}
+
+		sigParsed, err := schnorr.ParseSignature(in.Signature)
+		if err != nil {
+			return nil, fmt.Errorf("can't parse Schnorr "+
+				"signature: %v", err)
+		}
+
+		digest := chainhash.HashB(in.Msg)
```

Comment:
Doesn't this need to conditionally double hash based on the msg expectation?

Reply:
https://github.com/lightningnetwork/lnd/pull/6663#issuecomment-1170282963

Reply:
I don't think that's necessary. The double hashing is only needed by the remote signing setup (for signing a channel or node announcement) but we seem to be verifying it in a different way (not through the signer interface).